src/ui/actions/session.ts

```diff
@@ -205,7 +205,15 @@ export function createSocket(
         ),
         disableProtocolQueryCache: userData.get("backend_disableProtocolQueryCache"),
       };
-      if (userData.get("backend_newControllerOnRefresh")) {
+
+      const restartParam = new URL(window.location.href).searchParams.get("restart") || undefined;
```

Comment:
Nit: `URLSearchParams.get()` returns `null` for values not in the URL, so `|| undefined` isn't really doing anything meaningful.

```suggestion
      const restartParam = new URL(window.location.href).searchParams.get("restart");
```